pkg/query-service/featureManager/manager.go

```diff
@@ -3,32 +3,59 @@ package featureManager
 import (
 	"go.signoz.io/signoz/pkg/query-service/constants"
 	"go.signoz.io/signoz/pkg/query-service/model"
+	"go.uber.org/zap"
 )
 
 type FeatureManager struct {
-	activeFeatures model.FeatureSet
 }
 
 func StartManager() *FeatureManager {
-	fM := &FeatureManager{
-		activeFeatures: constants.DEFAULT_FEATURE_SET,
-	}
+	fM := &FeatureManager{}
 	return fM
 }
 
 // CheckFeature will be internally used by backend routines
 // for feature gating
 func (fm *FeatureManager) CheckFeature(featureKey string) error {
-	if value, ok := fm.activeFeatures[featureKey]; ok {
-		if value {
-			return nil
+
+	features, err := fm.GetFeatureFlags()
```

Comment:
Maybe implement and get the feature by the name `GetFeature(featureKey)`

Reply:
It is there: https://github.com/SigNoz/signoz/blob/22f4062aa3c60887932e6e17d5ab1cc8628402e0/pkg/query-service/featureManager/manager.go#LL58C1-L58C1
Will implement the function.